### DISCUSSION TITLE
Accept ProxyProtocol and Normal Connenction at same time

### DISCUSSION THREAD

**Discussion by APT-ZERO** [XTLS/Xray-core#3815] [Ideas]
Currently when i enable acceptProxyProtocol for an Inbound, it will not accept normal connections anymore
As ProxyProtocol both versions have static header in their starting packet, why can not accept ProxyProtocol  and Normal Connenction both at same time?
https://seriousben.com/posts/2020-02-exploring-the-proxy-protocol/#proxy-protocol-version-1

**Comment by mmmray** [comment]
if you expose xray to the internet in this mode, any user can forge their own IP address that shows in your logs. this is the same issue as reading x-forwarded-for even though there is no reverse proxy running (already an issue with the existing CDN transports, luckily everybody uses CDN with them... right?).

but more importantly, it's a problem for active probing. such a behavior can be easily distinguished from a normal website.

**Reply by APT-ZERO** [reply]
You are right
I wish we could listen to multiple addresses, one of them for normal connection on 0.0.0.0:PORT and another for proxyprotocol on localhost:PORT
